.clang-format

```diff
@@ -0,0 +1,38 @@
+---
+Language: Cpp
+AlignEscapedNewlinesLeft: 'true'
+AlignTrailingComments: 'true'
+AllowShortBlocksOnASingleLine: 'false'
+AllowShortFunctionsOnASingleLine: None
+AllowShortIfStatementsOnASingleLine: 'false'
+AllowShortLoopsOnASingleLine: 'false'
+ColumnLimit: '0'
+ContinuationIndentWidth: '4'
+IndentWidth: '4'
+SortIncludes: true
+IncludeCategories:
+  - Regex:           '<[[:alnum:].]+>'
+    Priority:        -1
+  - Regex:           '\".*\"'
+    Priority:        1
+IndentCaseLabels: false
+MaxEmptyLinesToKeep: '1'
+PointerAlignment: Left
```

Comment:
A lot of code in Overviewer appears to use right-aligned pointers. My C knowledge isn't the best, but I think

```C
int *foo, bar;
```

makes `foo` an int pointer and `bar` an int, whereas

```C
int* foo, bar;
```

makes both of them int pointers, correct?

In that case, wouldn't right-aligned pointers be the lesser trap?

Feel free to change my opinion on this, I'm mostly just noticing a lot of `foo *bar` to `foo* bar` transformations in this PR which have me a little worried over whether this was 100% intentional.

Reply:
Nope, both of those declare `foo` as a pointer and `bar` as an int. Yeah, it's pretty bad.

I used to use right-pointers because morally, I think pointerness should be part of the type. But C's syntax is such that right-pointers will get you in to trouble, like the confusion above, so nowadays I try to stick to left-pointers. I think that's a good choice for overviewer.

Reply:
So: I think I swapped left and right here... 😄 

I meant, the `int *foo` style is probably more appropriate. I don't know enough about `.clang-format` to know if that's what's selected here.

Reply:
I'm gonna be a vote for left-pointers as defining a mix of ints and int-pointers on the same line is kind of a weird basis for having it all be on the right through-out the codebase. Would rather keep all type and type modifiers next to the type, with the name on its own(`static volatile void* pointy`).